### PR TITLE
[Merged by Bors] - feat(Analysis/Analytic): `FormalMultilinearSeries.unshift` convergence

### PR DESCRIPTION
**Pull request by vasnesterov**
* Prove `unshift_shift`, which states that `p.unshift.shift = p`.  
* Prove `radius_shift` and `radius_unshift`, which state that the shifted and unshifted series converge on the same ball.  
* Prove `HasFPowerSeriesOnBall.unshift` (and variations), describing where the shifted series converges.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Analytic/Constructions.lean [hunk 1]
```diff
@@ -942,3 +942,47 @@ theorem Finset.analyticOnNhd_prod {A : Type*} [NormedCommRing A] [NormedAlgebra
...
+      rw [FormalMultilinearSeries.radius_unshift]
+      exact hf.r_le
+    r_pos := hf.r_pos
+    hasSum := by
+      intro y hy h'y
+      apply HasSum.zero_add
+      simp only [FormalMultilinearSeries.unshift, Nat.succ_eq_add_one,
+        continuousMultilinearCurryRightEquiv_symm_apply', add_sub_cancel_left]
+      exact (ContinuousLinearMap.apply 𝕜 F y).hasSum (hf.hasSum hy h'y)
+  }
```
Comment (sgouezel) [inline Mathlib/Analysis/Analytic/Constructions.lean:967]: can you use the `where` syntax here and below, like:
```suggestion
    HasFPowerSeriesWithinOnBall (fun y ↦ z + f y (y - x)) (pf.unshift z) s x r where
  r_le := by
    rw [FormalMultilinearSeries.radius_unshift]
    exact hf.r_le
  r_pos := hf.r_pos
  hasSum := by
    intro y hy h'y
    apply HasSum.zero_add
    simp only [FormalMultilinearSeries.unshift, Nat.succ_eq_add_one,
      continuousMultilinearCurryRightEquiv_symm_apply', add_sub_cancel_left]
    exact (ContinuousLinearMap.apply 𝕜 F y).hasSum (hf.hasSum hy h'y)
```

Comment (vasnesterov) [reply Mathlib/Analysis/Analytic/Constructions.lean]: Done. Should I also make similar changes to other definitions in this file? For example, `HasFPowerSeriesWithinOnBall.add`.

Comment (sgouezel) [reply Mathlib/Analysis/Analytic/Constructions.lean]: Perfect, thanks. No need to change the other ones, it's out of scope for this PR.

**Review by sgouezel — COMMENTED** [review]
bors r+
Thanks!

**Pull request closed without merging**